File: backend/src/main/java/z9/hobby/domain/classes/repository/ClassUserRepository.java
package z9.hobby.domain.classes.repository;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import z9.hobby.domain.classes.entity.ClassEntity;
import z9.hobby.domain.classes.entity.ClassUserEntity;

public interface ClassUserRepository extends JpaRepository<ClassUserEntity, Long> {
    boolean existsByUserIdAndClassesId(Long userId, Long classId);
    boolean existsByClasses_IdAndUserId(Long classId, Long userId);
    Optional<ClassUserEntity> findByClassesIdAndUserId(Long classId, Long userId);

    void deleteByUserId(Long userId);

    List<ClassUserEntity> findByClassesId(Long classId);

    // 특정 모임에서 특정 유저(모임장)를 제외한 회원 수 조회
    long countByClassesIdAndUserIdNot(Long classId, Long userId);

    boolean existsByClassesAndUserId(ClassEntity classEntity, Long userId);
}

File: backend/src/main/java/z9/hobby/domain/classes/repository/ClassUserRepository.kt
package z9.hobby.domain.classes.repository

import org.springframework.data.jpa.repository.JpaRepository
import z9.hobby.domain.classes.entity.ClassEntity
import z9.hobby.domain.classes.entity.ClassUserEntity
import java.util.*

interface ClassUserRepository : JpaRepository<ClassUserEntity, Long> {
    fun existsByUserIdAndClassesId(userId: Long, classId: Long): Boolean
    fun existsByClasses_IdAndUserId(classId: Long, userId: Long): Boolean
    fun findByClassesIdAndUserId(classId: Long, userId: Long): Optional<ClassUserEntity>

    fun deleteByUserId(userId: Long)

    fun findByClassesId(classId: Long): List<ClassUserEntity>

    // 특정 모임에서 특정 유저(모임장)를 제외한 회원 수 조회
    fun countByClassesIdAndUserIdNot(classId: Long, userId: Long): Long

    fun existsByClassesAndUserId(classEntity: ClassEntity, userId: Long): Boolean
}
